includes/class-custom-fonts-api-init.php

```diff
@@ -0,0 +1,216 @@
+	}
+
+	/**
+	 * Register API routes.
+	 *
+	 * @since x.x.x
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			$this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_admin_settings' ),
+					'permission_callback' => array( $this, 'get_permissions_check' ),
+					'args'                => array(),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Get common settings.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return array $updated_option defaults + set DB option data.
+	 *
+	 * @since x.x.x
+	 */
+	public function get_admin_settings( $request ) {
+
+		$args = array(
+			'post_type'      => BSF_CUSTOM_FONTS_POST_TYPE,
+			'post_status'    => 'publish',
+			'orderby'        => 'ID',
+			'posts_per_page' => -1,
+		);
+
+		$search_query = sanitize_text_field( $request->get_param( 'q' ) );
+		if ( isset( $search_query ) && ! empty( $search_query ) && is_string( $search_query ) ) {
+			$args['s'] = $search_query;
+		}
+
+		$bsf_fonts_data        = array();
+		$query_posts           = new WP_Query( $args );
+		$bsf_custom_font_posts = wp_count_posts( BSF_CUSTOM_FONTS_POST_TYPE );
+
+		if ( $query_posts && $query_posts->have_posts() ) {
+			while ( $query_posts->have_posts() ) {
+				$query_posts->the_post();
+				global $post;
+
+				$font_post_data = array(
+					'id'         => $post->ID,
+					'title'      => $post->post_title,
+					'slug'       => $post->post_name,
+					'fonts-face' => get_post_meta( $post->ID, 'fonts-face', true ),
+					'font-type'  => get_post_meta( $post->ID, 'font-type', true ),
+					'fonts-data' => get_post_meta( $post->ID, 'fonts-data', true ),
+				);
+
+				$bsf_fonts_data[] = $font_post_data;
+				wp_reset_postdata();
+			}
+		}
+
+		$defaults = apply_filters(
+			'bsf_custom_fonts_rest_data',
+			array(
+				'fonts'              => $bsf_fonts_data,
+				'found_posts'        => $query_posts->found_posts,
+				'active_fonts_count' => isset( $bsf_custom_font_posts->publish ) ? intval( $bsf_custom_font_posts->publish ) : 0,
+				'trash_fonts_count'  => isset( $bsf_custom_font_posts->trash ) ? intval( $bsf_custom_font_posts->trash ) : 0,
+				'draft_fonts_count'  => isset( $bsf_custom_font_posts->draft ) ? intval( $bsf_custom_font_posts->draft ) : 0,
+			)
+		);
+
+		return $defaults;
+	}
+
+	/**
+	 * Check whether a given request has permission to read notes.
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|boolean
+	 * @since x.x.x
+	 */
+	public function get_permissions_check( $request ) {
+
+		if ( ! current_user_can( 'edit_theme_options' ) ) {
+			// return new WP_Error( 'bsf_custom_fonts_rest_cannot_view', __( 'Sorry, you cannot list resources.', 'custom-fonts' ), array( 'status' => rest_authorization_required_code() ) );
```

Comment:
Commented code. If not required please remove it @imnavanath

Reply:
Updated, for debugging I was commented bu removed comment now, this is valid code to restrict rest API visit.